thirdparty/runtime_environments_test.go

```diff
@@ -35,3 +35,56 @@ func TestGetOSInfo(t *testing.T) {
 	require.NoError(t, err)
 	assert.Len(result, 10)
 }
+
+func TestGetToolchains(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	assert := assert.New(t)
+	config := testutil.TestConfig()
+	testutil.ConfigureIntegrationTest(t, config, "TestGetToolchains")
+	c := NewRuntimeEnvironmentsClient(config.RuntimeEnvironments.BaseURL, config.RuntimeEnvironments.APIKey)
+
+	// Verify that there are no errors with ToolchainFilterOptions including the AMI and limit.
+	ami := "ami-016662ab459a49e9d"
+	opts := ToolchainFilterOptions{
+		AMIID: ami,
+		Limit: 10,
+	}
+	result, err := c.getToolchains(ctx, opts)
+	require.NoError(t, err)
+	assert.Len(result, 10)
+
+	// Verify that we filter correctly by name and version.
+	name := "nodejs"
+	version := "toolchain_version_v16.17.0"
+	opts = ToolchainFilterOptions{
+		AMIID:   ami,
+		Page:    0,
+		Limit:   5,
+		Name:    name,
+		Version: version,
+	}
+	result, err = c.getToolchains(ctx, opts)
+	require.NoError(t, err)
+	require.NotEmpty(t, result)
+	assert.Equal(result[0].Name, name)
+	assert.Equal(result[0].Version, version)
+	assert.Len(result, 1)
+
+	// Verify that we receive no results for a fake toolchain
+	name = "blahblahblah"
+	opts = ToolchainFilterOptions{
+		AMIID: ami,
+		Page:  0,
+		Limit: 5,
+		Name:  name,
+	}
```

Comment:
No need to declare a variable if it's once in a test! Also maybe consider removing page/limit for this one
```suggestion
	opts = ToolchainFilterOptions{
		AMIID: ami,
		Page:  0,
		Limit: 5,
		Name:  "blahblahblah",
	}
```